manifests/repo.pp

```diff
@@ -1,4 +1,56 @@
+# This class installs the hashicorp repository
 #
-class hashi_stack::repo () {
-  
-}
\ No newline at end of file
+# @summary Set up the package repository for the HashiCorp Stack components
+#
+# @example
+#   include hashi_stack::repo
+#
+# @param priority A numeric priority for the repo, passed to the package management system
+# @param proxy The URL of a HTTP proxy to use for package downloads (YUM only)
+# @param base_repo_url The base url for the repo path
+class hashi_stack::repo (
+  Optional[Integer] $priority      = undef,
+  String            $proxy         = 'absent',
+) {
+  $key_id='E8A032E094D8EB4EA189D270DA418C88A3219F7B'
+  $key_source='https://apt.releases.hashicorp.com/gpg'
+  $description='HashiCorp package repository.'
```

Comment:
Why aren't these parameters? In my experience GPG keys can change and it's nice to be able to override it.

Reply:
i ll make those changes!